cluster/kubernetes.go

```diff
@@ -202,3 +202,18 @@ func (b *KubeCluster) ValidateCreationFields(r *components.CreateClusterRequest)
 func (b *KubeCluster) GetSecretWithValidation() (*secret.SecretsItemResponse, error) {
 	return b.commonSecret.get(b)
 }
+
+// SaveConfigSecretId saves the config secret id in database
+func (b *KubeCluster) SaveConfigSecretId(configSecretId string) error {
```

Comment:
Apparently the `SaveConfigSecretId`, `GetConfigSecretId` is replicated for each kubernetes types. If possible this should be factored or use composition to avoid copy pasting the same code to multiple places

Reply:
Unfortunately, this means a big refactor. I did an issue for this: #539